function foo returns int
    a = 1
    try
        b = 2
        return b
    catch
        c = 3
        return c
    return a
